frontend/apps/web/app/(mgmt)/[account]/runs/[id]/page.tsx

```diff
@@ -108,6 +108,43 @@ export default function Page({ params }: PageProps): ReactElement {
   const [isRetrievingRunContext, setIsRetrievingRunContext] =
     useState<boolean>(false);
 
+  const [duration, setDuration] = useState<string>('');
+
+  useEffect(() => {
+    let timer: NodeJS.Timeout;
+    if (jobRun?.startedAt && jobRun?.status === JobRunStatusEnum.RUNNING) {
+      const updateDuration = () => {
+        const now = new Date();
+        const startTime = jobRun?.startedAt?.toDate();
+        if (startTime) {
+          const diffInSeconds = Math.floor(
+            (now.getTime() - startTime.getTime()) / 1000
+          );
+          const minutes = Math.floor(diffInSeconds / 60);
+          const seconds = diffInSeconds % 60;
+          // handle plural minutes and seconds and set the duration
+          setDuration(
+            minutes > 0
+              ? `${minutes} minute${minutes > 1 ? 's' : ''} ${seconds} second${seconds !== 1 ? 's' : ''}`
+              : `${seconds} second${seconds !== 1 ? 's' : ''}`
+          );
+        }
```

Comment:
I think you got the logic right in the else branch. `setDuration(getDuration(end, start))`
Isn't the end in this case `const now = new Date()` ?
Not sure there is really a need to duplicate the logic that is already written in the `getDuration` method.

Reply:
makes sense, will clean it up